### PR TITLE
add APIGW NextGen doc

### PR DESCRIPTION
**Pull request by bentsku**
This PR adds documentation about the NextGen API provider. Community provider work can be seen here: https://github.com/localstack/localstack/pull/11035

# TODO

- [ ] update the link to `3.8.0`

### CODE REVIEW

File: content/en/user-guide/aws/apigateway/index.md [hunk 1]
```diff
@@ -206,6 +206,42 @@ $ curl -X GET http://localhost:4566/restapis/<REST_API_ID>/test/_user_request_/t
 {"message":"Hello World"}
 {{< /command >}}
 
+## New API Gateway implementation
+
+{{< callout >}}
+
+Since [3.8.0](link here), LocalStack supports a new API Gateway implementation for both API Gateway v1 (REST API) and v2 (HTTP API).
```
Comment (cloutierMat) [inline content/en/user-guide/aws/apigateway/index.md:213]: Just a reminder to not forget to update this link before merging. I also added a TODO on the PR comment

**Review by cloutierMat — APPROVED** [review]
Thanks for updating the docs! 

I think you cover all of the important points! So much work described in a few sentences.

We should update the link to the configuration page to point to the right section of the doc. But other that that! :+1:

File: content/en/user-guide/aws/apigateway/index.md [hunk 2]
```diff
@@ -206,6 +206,42 @@ $ curl -X GET http://localhost:4566/restapis/<REST_API_ID>/test/_user_request_/t
...
+
+You can [set the following flag]({{< ref "configuration" >}}) `PROVIDER_OVERRIDE_APIGATEWAY=next_gen` to use the new implementation.
```
Comment (bentsku) [inline content/en/user-guide/aws/apigateway/index.md:215]:
```suggestion
You can [set the following flag]({{< ref "configuration#api-gateway" >}}) `PROVIDER_OVERRIDE_APIGATEWAY=next_gen` to use the new implementation.
```

**Review by HarshCasper — APPROVED** [review]
Merging this now — I will add the link later

**Pull request merged**